frontend/src/Explorer/View/Search.purs

```diff
@@ -0,0 +1,159 @@
+module Explorer.View.Search
+    ( searchInputView
+    , searchItemViews
+    ) where
+
+import Prelude
+import Data.Lens ((^.))
+import Data.Maybe (Maybe(..))
+import Data.String (length)
+import Data.Tuple (Tuple(..))
+import Explorer.I18n.Lang (Language, translate)
+import Explorer.I18n.Lenses (cAddress, cEpoch, cSlot, cTransaction, common, hero, hrSearch, hrTime) as I18nL
+import Explorer.Lenses.State (gViewSearchInputFocused, globalViewState, lang, gViewSearchQuery, gViewSearchTimeQuery, gViewSelectedSearch, viewStates)
+import Explorer.State (maxSlotInEpoch)
+import Explorer.Types.Actions (Action(..))
+import Explorer.Types.State (Search(..), State)
+import Pux.Html (Html, div, text, ul, li, label, input) as P
+import Pux.Html.Attributes (checked, className, htmlFor, id_, maxLength, name, type_, placeholder, value) as P
+import Pux.Html.Events (onChange, onClick, onFocus, onBlur, onKey) as P
+
+inputEpochName :: String
+inputEpochName = "inp_epoch"
+
+inputSlotName :: String
+inputSlotName = "inp_slot"
+
+searchInputView :: State -> P.Html Action
+searchInputView state =
+    let lang' = state ^. lang
+        dbViewSearchInputFocused = state ^. (viewStates <<< globalViewState <<< gViewSearchInputFocused)
+        searchIconClazz = if dbViewSearchInputFocused then " bg-icon-search-hover" else " bg-icon-search"
+        selectedSearch = state ^. (viewStates <<< globalViewState <<< gViewSelectedSearch)
+        addrHiddenClazz = if selectedSearch == SearchTime  then " hide " else ""
+        epochHiddenClazz = if selectedSearch /= SearchTime  then " hide " else ""
+        focusedClazz = if dbViewSearchInputFocused then " focused " else ""
+        searchTimeQuery = state ^. (viewStates <<< globalViewState <<< gViewSearchTimeQuery)
+    in
+    P.div
+        [ P.className $ "explorer-search__container" <> focusedClazz ]
+        [ P.input
+            [ P.className $ "explorer-search__input explorer-search__input--address-tx"
+                          <> addrHiddenClazz
+                          <> focusedClazz
+            , P.type_ "text"
+            , P.placeholder $ if dbViewSearchInputFocused
+                              then ""
+                              else translate (I18nL.hero <<< I18nL.hrSearch) lang'
+            , P.onFocus <<< const $ GlobalFocusSearchInput true
+            , P.onBlur <<< const $ GlobalFocusSearchInput false
+            , P.onChange $ GlobalUpdateSearchValue <<< _.value <<< _.target
+            , P.onKey "enter" $ const GlobalSearch
+            , P.value $ state ^. (viewStates <<< globalViewState <<< gViewSearchQuery)
+            ]
+            []
+        , P.div
+            [ P.className $ "explorer-search__wrapper" <> epochHiddenClazz ]
```

Comment:
For example, from this line it is not clear did we accidentally concatenate these two classes. With `String.joinWith " " [...]` I would be sure and won't have to look to classes to see do they have extra space or not.